Makefile

```diff
@@ -1,14 +1,21 @@
-.PHONY: all
+MAKE_VARIABLES := $(.VARIABLES)
 
 GO_VERSION=$(shell go version | cut -d ' ' -f 3 | sed -e 's/ /-/g' | sed -e 's/\//-/g' | sed -e 's/^go//g')
-GO_PLATFORM ?= $(go version | cut -d ' ' -f 4 | sed -e 's/ /-/g' | sed -e 's/\//-/g')
+GO_PLATFORM ?= $(shell go version | cut -d ' ' -f 4 | sed -e 's/ /-/g' | sed -e 's/\//-/g')
 BEATS_VERSION ?= "master"
 BEATS_TAG ?= $(shell echo ${BEATS_VERSION} | sed 's/[^[:digit:]]*\([[:digit:]]*\(\.[[:digit:]]*\)\)/v\1/')
 AWSBEATS_VERSION ?= "1-snapshot"
 DOCKER_IMAGE ?= s12v/awsbeats
 DOCKER_TAG ?= canary
 
-all: test beats build
+.PHONY: all
+all: vars test beats build
```

Comment:
We now dump all make variables defined in this Makefile for traceability.
It will help us debugging issues like this in future 😉 

```console
$ make
make variables for this build:
DOCKER_TAG = canary
GO_PLATFORM = darwin-amd64
AWSBEATS_VERSION = "1-snapshot"
GO_VERSION = 1.10.2
DOCKER_IMAGE = s12v/awsbeats
BEATS_TAG = master
BEATS_VERSION = "master"
To override, run make like `make SOME_VAR=some_value`.
test -z "$(find . -path ./vendor -prune -type f -o -name '*.go' -exec gofmt -d {} + | tee /dev/stderr)"
*snip*
```